src/create_confirmation_request.cc

```diff
@@ -0,0 +1,112 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
we should really avoid this pattern of putting everything in the `src` root, it's going to cause issues with conflicting header include paths.

Reply:
Totally agree as we discussed before, can we do this after 1.0 as we need to do this across all native libs?